tests/test_filetools_txt.py

```diff
@@ -18,6 +20,37 @@ def test_check_encoding_of_file():
         assert check_encoding_of_file(PATH_TO_TEST_OUTPUT_TXT_FILE) == encoding
         PATH_TO_TEST_OUTPUT_TXT_FILE.unlink()
 
+@pytest.mark.parametrize(
+    "file_stem, line_number_to_cut, line_number_to_insert_before",
+    [
+        ("tc1-2", 1, 2),
+        ("tc1-2", 2, 2),
+        ("tc3-4", 3, 2),
+        ("tc3-4", 2, 4),
+        ("tc5", 1, 5),
+        ("tc6", 4, 1),
+        ("tc7", 3, "END"),
+        ("tc8", 3, 0),
+        ("tc9-11", 0, "END"),
+        ("tc9-11", 0, "end"),
+        ("tc9-11", 0, "FINISH"),
```

Comment:
```suggestion
        ("tc9-11", 0, "END"),
        ("tc9-11", 0, "end"),
```

As we discussed, `("tc9-11", 0, "FINISH")` cannot be in `pytest.mark.parametrize` along with other cases, because we expect a totally different behaviour. You need to create a different test function and use `pytest.raises()` to catch a specific error with a specific error message.

It could be called, e.g.:

```py
def test_move_line_throws_exception_with_wrong_string()
```